pallet/src/lib.rs

```diff
@@ -116,6 +154,23 @@ pub mod pallet {
         }
     }
 
+    impl<T: Config> SessionManager<T::AccountId> for AlephSessionManager<T> {
+        fn new_session(session: u32) -> Option<Vec<T::AccountId>> {
+            if let Some((validators_list, validators_list_session)) = Pallet::<T>::validators_list()
```

Comment:
While fetching "the whole `validators_list`" might sounds innocuous, in reality what is stored in the database and needs to be fetched (from disk) is `20-100` times larger then the original data.

Reply:
I believe that overhead is related to the number of calls not the size of single element, but I'm not 100% sure.

Reply:
The rule of thumb in designing storage is to bundle several items into structs (or pair like here) only if you access and modify the fields "together". According to my understanding this is not quite the case here -- you would use the session number many times and the validators list only once.

Now I think that the "Vec" type data would be stored under a single key, so the additional cost here would only be fetching the validators list from disk (so the `20-100` times overhead is probably not quite correct -- it would be correct when talking about a single scalar storage value). But this is still not cheap, as it is serialized, hashed and whatnot.

Reply:
Hashed is only the key to the db, but the vector is indeed deserialized from bytes, so you are right that in this case it's better to store separately.

Reply:
Done